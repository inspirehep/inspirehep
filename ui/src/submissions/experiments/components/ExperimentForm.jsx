import React from 'react';
import { Field, Form } from 'formik';
import { Row, Col } from 'antd';

import SelectField from '../../common/components/SelectField';
import SubmitButton from '../../common/components/SubmitButton';
import SuggesterField from '../../common/components/SuggesterField';
import { EXPERIMENT_TYPE_OPTIONS } from '../schemas/constants';

const ExperimentForm = () => {
  const getSuggestionSourceLegacyName = (suggestion) =>
    suggestion._source.legacy_name;

  return (
    <Form className="bg-white pa3">
      <Row className="pt3 bg-white">
        <Col span={24}>
          <Field
            name="project_type"
            label="* Experiment type"
            mode="multiple"
            options={EXPERIMENT_TYPE_OPTIONS}
            placeholder="Select types"
            component={SelectField}
          />
        </Col>
      </Row>

      <Row className="mb3 pt2 bg-white">
        <Col span={24}>
          <Field
            component={SuggesterField}
            extractItemCompletionValue={getSuggestionSourceLegacyName}
            label="* Legacy name"
            name="legacy_name"
            pidType="experiments"
            placeholder="Legacy name, type for suggestions"
            searchasyoutype="true"
            suggesterName="experiment"
          />
        </Col>
      </Row>

      <Row type="flex" justify="end">
        <SubmitButton />
      </Row>
    </Form>
  );
};

export default ExperimentForm;
