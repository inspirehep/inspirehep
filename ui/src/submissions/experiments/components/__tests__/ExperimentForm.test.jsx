import React from 'react';
import { shallow } from 'enzyme';

import ExperimentForm from '../ExperimentForm';

describe('ExperimentForm', () => {
  it('renders', () => {
    const wrapper = shallow(<ExperimentForm />);
    expect(wrapper).toMatchSnapshot();
  });
});
