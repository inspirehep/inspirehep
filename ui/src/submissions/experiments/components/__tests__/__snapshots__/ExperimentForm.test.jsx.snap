// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperimentForm renders 1`] = `
<Form
  className="bg-white pa3"
>
  <Row
    className="pt3 bg-white"
  >
    <Col
      span={24}
    >
      <Field
        component={[Function]}
        label="* Experiment type"
        mode="multiple"
        name="project_type"
        options={
          Array [
            Object {
              "display": "Collaboration",
              "value": "collaboration",
            },
            Object {
              "display": "Experiment",
              "value": "experiment",
            },
            Object {
              "display": "Accelerator",
              "value": "accelerator",
            },
          ]
        }
        placeholder="Select types"
      />
    </Col>
  </Row>
  <Row
    className="mb3 pt2 bg-white"
  >
    <Col
      span={24}
    >
      <Field
        component={[Function]}
        extractItemCompletionValue={[Function]}
        label="* Legacy name"
        name="legacy_name"
        pidType="experiments"
        placeholder="Legacy name, type for suggestions"
        searchAsYouType={true}
        suggesterName="experiment"
      />
    </Col>
  </Row>
  <Row
    justify="end"
    type="flex"
  >
    <SubmitButton />
  </Row>
</Form>
`;
